specs/phase1/beacon-chain.md

```diff
@@ -886,9 +886,10 @@ def process_crosslink_for_shard(state: BeaconState,
         for attestation in transition_attestations:
             participants = get_attesting_indices(state, attestation.data, attestation.aggregation_bits)
             transition_participants = transition_participants.union(participants)
-            assert attestation.data.shard_head_root == shard_transition.shard_data_roots[
-                len(shard_transition.shard_data_roots) - 1
-            ]
+            if len(shard_transition.shard_data_roots) > 0:
+                # Is the `shard_transition` candidate
+                last_offset_index = len(shard_transition.shard_data_roots) - 1
+                assert attestation.data.shard_head_root == shard_transition.shard_data_roots[last_offset_index]
```

Comment:
Is this still correct? I was under the impression that `shard_data_roots` is used for shard block bodies. Ex:
```python
hash_tree_root(shard_block.message.body)
```

https://github.com/ethereum/eth2.0-specs/issues/1881

Reply:
Ahh yes, sorry I forgot it when I fixing another issue!

So here this line should be

`assert attestation.data.shard_head_root == shard_transition.shard_states[last_offset_index].latest_block_root`

right?

Reply:
That looks good to me!